tools/serverpod_cli/lib/src/commands/create.dart

```diff
@@ -89,6 +90,22 @@ class CreateCommand extends ServerpodCommand<CreateOption> {
       throw ExitException.error();
     }
 
+    // Make sure all necessary downloads are installed
+    if (!resourceManager.isTemplatesInstalled) {
+      try {
+        await resourceManager.installTemplates();
+      } catch (e) {
+        log.error('Failed to download templates.');
+        throw ExitException.error();
+      }
+
+      if (!resourceManager.isTemplatesInstalled) {
```

Comment:
Is this needed?

I think `installTemplate` already errors on each error, right?

The bigger problem (unrelated to this move) is that `isTemplatesInstalled` just checks the existence of the dir, not if the dir was ever written to successfully. 

Likewise `isTemplatesInstalled` does nothing if the dir exists already.

So I think in case the user is offline during the first run, this'll be broken "forever" (on that version).

Probably a better approach would be to use a temp download dir and then move that in the end (if that can be done atomically), or at least only create the dir when the file is received, so that the unzipping very likely succeeds.

Reply:
Created https://github.com/serverpod/serverpod/issues/3790 to track this issue. I'll pick it up soon